components/modal.vue

```diff
@@ -1,11 +1,11 @@
 <template>
     <transition :name="'v-modal-' + transitionName">
         <div
-            @click.stop="close('outer')"
+            @click.self="close('outer')"
             class="v-modal__mask"
             v-if="show">
-            <div class="v-modal__wrapper" :style="theme">
-                <div @click.stop class="v-modal__content" :class="size" :id="id" :style="style">
+            <div @click.self="close('outer')" class="v-modal__wrapper" :style="theme">
```

Comment:
This doesn't really make sense. The `@click.stop` was to prevent propagation. It was never used to close the modal.

Reply:
Actually the click.stop was preventing other unrelated events within the modal content. By moving the event to the wrapper from the content and only tracking itself, the content won't close on click, but the wrapper will still close successfully, hopefully that made sense?

Reply:
but the event is already tracked on the mask. Isn't that enough?

Reply:
No unfortunately, tried that first and it didn't work. They have weird overlaps and using the .self modifier will restrict the event to that element and exclude the children